src/screens/TopicPageScreen/elements/TopicDomainHeader.js

```diff
@@ -1,38 +1,66 @@
+import CommunityIcon from '../../../assets/icon/CommunityIcon';
+import {Shimmer} from '../../../components/Shimmer/Shimmer';
+import {colors} from '../../../utils/colors';
 import {fonts, normalize, normalizeFontSize} from '../../../utils/fonts';
 import {convertString} from '../../../utils/string/StringUtils';
-import {colors} from '../../../utils/colors';
 
 const TopicDomainHeader = (props) => {
   const {domain, handleOnMemberPress, hideSeeMember, isFollow, memberCount} = props;
+  const isUserFollow = props?.initialData?.isFollowing ? props.initialData.isFollowing : isFollow;
+
   const handlePress = () => {
-    if (isFollow) {
+    if (isUserFollow) {
       handleOnMemberPress();
     } else {
       SimpleToast.show('Only community members can see other members', SimpleToast.SHORT);
     }
   };
-
   return (
-    <View>
-      <Text style={styles.domainText} numberOfLines={1} ellipsizeMode="tail">
+    <View
+      style={{
+        alignContent: 'flex-start',
+        alignItems: 'flex-start'
+      }}>
+      <Text
+        style={styles.domainText(props.isHeaderHide || props.hasSearch)}
+        numberOfLines={1}
+        ellipsizeMode="tail">
         {`#${convertString(domain, ' ', '')}`}
       </Text>
       <Pressable onPress={handlePress} style={{backgroundColor: 'transparent'}}>
-        <View style={{flexDirection: 'row', alignItems: 'center'}}>
-          <FastImage testID="imageTopicMember" source={TopicMemberIcon} style={styles.member} />
-          <Text style={styles.domainMember}>{memberCount} Members</Text>
+        <View style={{flexDirection: 'row', alignItems: 'center', marginTop: normalize(1)}}>
+          <CommunityIcon
+            color={props.isHeaderHide || props.hasSearch ? '#FFFFFF' : undefined}
+            style={{
+              marginRight: normalize(5),
+              height: normalize(8)
+            }}
+          />
+          {props?.initialData?.memberCount === undefined && props.isLoading ? (
+            <Shimmer height={10} width={normalize(25)} />
+          ) : (
+            <Text style={styles.domainMember(props.isHeaderHide || props.hasSearch)}>
+              {props?.initialData?.memberCount || memberCount}
+            </Text>
+          )}
+          <Text style={styles.domainMember(props.isHeaderHide || props.hasSearch)}> Members</Text>
         </View>
-        {isFollow && !hideSeeMember && (
-          <Text style={styles.seeMemberText} numberOfLines={1} ellipsizeMode="tail">
-            See community member
-          </Text>
+        {props?.initialData?.memberCount === undefined && props.isLoading ? (
+          <Shimmer height={10} width={normalize(60)} />
+        ) : (
+          isUserFollow &&
+          !hideSeeMember && (
+            <Text
+              style={styles.seeMemberText(props.isHeaderHide)}
+              numberOfLines={1}
+              ellipsizeMode="tail">
+              See community members
+            </Text>
+          )
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a `Shimmer` component to display while loading data, which is a good practice for improving user experience. However, there's a potential issue with the conditional rendering logic in lines 43-63. If `props?.initialData?.memberCount` is `0`, it will be considered as `undefined` due to JavaScript's falsy values, and the `Shimmer` component will be displayed even if the data has already been loaded. To fix this, you should explicitly check for `undefined` instead of relying on truthy/falsy values.
```diff
- {props?.initialData?.memberCount === undefined && props.isLoading ? (
+ {props?.initialData?.memberCount === undefined && props.isLoading !== false ? (
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->